include/clang/AST/Expr.h

```diff
@@ -2962,6 +2962,77 @@ class CStyleCastExpr final
   friend class CastExpr;
 };
 
+class BoundsCastExpr final
+    : public ExplicitCastExpr,
+      private llvm::TrailingObjects<BoundsCastExpr, CXXBaseSpecifier *> {
+  SourceLocation LPLoc;
+  SourceLocation RPLoc;
+  Expr *BaseExpr = nullptr;
```

Comment:
I don't think BaseExpr is necessary: BaseExpr is stored in the base class CastExpr.